src/java/org/apache/cassandra/net/MessagingService.java

```diff
@@ -668,7 +670,9 @@ public int sendRRWithFailure(MessageOut message, InetAddress to, IAsyncCallbackW
     public int sendRR(MessageOut message, InetAddress to, IAsyncCallback cb, long timeout, boolean failureCallback)
     {
         int id = addCallback(cb, message, to, timeout, failureCallback);
-        sendOneWay(failureCallback ? message.withParameter(FAILURE_CALLBACK_PARAM, ONE_BYTE) : message, id, to);
+        if (!testOnly) {
```

Comment:
why is this needed?

Reply:
Wasn't able to mock a remote address for testing and saw this pattern already used for a test instance of MessagingService. I just found that I can use 127.0.0.2 to model a remote endpoint so I'll remove all this